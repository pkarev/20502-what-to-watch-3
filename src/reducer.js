import movies from './mocks/movies.js';

const ALL_GENRES_FILTER = `All genres`;

const initialState = {
  activeGenreFilter: ALL_GENRES_FILTER,
  movies,
  genresList: [],
};

const ActionType = {
  SET_GENRES_FILTER: `SET_GENRES_FILTER`,
  SET_GENRES_LIST: `SET_GENRES_LIST`,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRES_FILTER:
      return (Object.assign({}, state, {
        activeGenreFilter: action.payload,
      }));
    case ActionType.SET_GENRES_LIST:
      return (Object.assign({}, state, {
        genresList: action.payload,
      }));
  }

  return state;
};

const ActionCreator = {
  setGenresFilter: (genreFilter) => ({
    type: ActionType.SET_GENRES_FILTER,
    payload: genreFilter,
  }),
  setGenresList: (uniqueGenres) => {
    return {
      type: ActionType.SET_GENRES_LIST,
      payload: uniqueGenres,
    };
  },
};

export {reducer, ActionCreator, ActionType, ALL_GENRES_FILTER};
