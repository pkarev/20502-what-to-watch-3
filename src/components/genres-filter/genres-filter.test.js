import React from 'react';
import renderer from 'react-test-renderer';
import GenresFilter from './genres-filter.jsx';
import {ALL_GENRES_FILTER} from '../../reducer';

const genresList = [ALL_GENRES_FILTER, `action`, `comedy`, `drama`];

it(`Render GenresFilter`, () => {
  const tree = renderer
    .create(<GenresFilter genresList={genresList} activeFilter="drama" onGenresFilterClick={() => {}}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
