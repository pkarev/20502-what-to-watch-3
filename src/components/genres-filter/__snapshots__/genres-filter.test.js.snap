// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresFilter 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      action
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      comedy
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      drama
    </a>
  </li>
</ul>
`;
